runelite-client/src/main/java/net/runelite/client/plugins/corp/CorpDamageOverlay.java

```diff
@@ -0,0 +1,135 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.corp;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.Varbits;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.ComponentConstants;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+
+public class CorpDamageOverlay extends Overlay
+{
+	private final Client client;
+	private final CorpPlugin corpPlugin;
+
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	@Inject
+	private CorpDamageOverlay(Client client, CorpPlugin corpPlugin)
+	{
+		setPosition(OverlayPosition.TOP_LEFT);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
+		setPriority(OverlayPriority.LOW);
+		this.client = client;
+		this.corpPlugin = corpPlugin;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		NPC corp = corpPlugin.getCorp();
+		if (corp == null)
+		{
+			return null;
+		}
+
+		Widget damageWidget = client.getWidget(WidgetID.CORP_DAMAGE, 0);
+		if (damageWidget != null)
+		{
+			damageWidget.setHidden(true);
+		}
+
+		int myDamage = client.getVar(Varbits.CORP_DAMAGE);
+		int totalDamage = corpPlugin.getTotalDamage();
+		int players = corpPlugin.getPlayers().size();
+
+		// estimate how much damage is required for kill based on number of players
+		int damageForKill = players != 0 ? totalDamage / players : 0;
+
+		panelComponent.getChildren().clear();
+
+		panelComponent.getChildren().add(TitleComponent.builder()
+			.text("Corporeal Beast")
```

Comment:
I feel like title is completely unnecessary, it is obvious that it is for corp when it shows only when fighting corp.